src/crypto/equihash.cpp

```diff
@@ -458,6 +472,7 @@ std::set<std::vector<eh_index>> Equihash<N,K>::OptimisedSolve(const eh_HashState
     std::set<std::vector<eh_index>> solns;
     eh_index recreate_size { UntruncateIndex(1, 0, CollisionBitLength + 1) };
     int invalidCount = 0;
```

Comment:
I can also see two other options to avoid the memory leak:
1. Wrap the whole thing in a try/catch that deletes the elements of ``partialSolns`` in the catch, and rethrows.
2. Use a container whose destructor deletes its elements for ``partialSolns``.